livekit-agents/livekit/agents/inference/tts.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import annotations
+
+import asyncio
+import base64
+import json
+import os
+import weakref
+from dataclasses import dataclass, replace
+from typing import Any, Literal
+
+import aiohttp
+
+from .. import tokenize, tts, utils
+from .._exceptions import APIConnectionError, APIError, APIStatusError, APITimeoutError
+from ..log import logger
+from ..types import DEFAULT_API_CONNECT_OPTIONS, NOT_GIVEN, APIConnectOptions, NotGivenOr
+from ..utils import is_given
+from ._utils import create_access_token
+from .models import TTSModels
+
+TTSEncoding = Literal[
+    "pcm_s16le",
+    # "pcm_mulaw",
+]
+
+DEFAULT_ENCODING: TTSEncoding = "pcm_s16le"
+DEFAULT_SAMPLE_RATE: int = 16000
+DEFAULT_BASE_URL = "https://agent-gateway.livekit.cloud/v1"
+
+
+@dataclass
+class _TTSOptions:
+    model: NotGivenOr[TTSModels | str]
+    voice: NotGivenOr[str]
+    language: NotGivenOr[str]
+    encoding: TTSEncoding
+    sample_rate: int
+    base_url: str
+    api_key: str
+    api_secret: str
+    extra_kwargs: dict[str, Any]
+
+
+class TTS(tts.TTS):
+    def __init__(
+        self,
+        *,
+        model: NotGivenOr[TTSModels | str] = NOT_GIVEN,  # TODO: add a default model
+        voice: NotGivenOr[str] = NOT_GIVEN,
+        language: NotGivenOr[str] = NOT_GIVEN,
+        encoding: NotGivenOr[TTSEncoding] = NOT_GIVEN,
+        sample_rate: NotGivenOr[int] = NOT_GIVEN,
+        base_url: NotGivenOr[str] = NOT_GIVEN,
+        api_key: NotGivenOr[str] = NOT_GIVEN,
+        api_secret: NotGivenOr[str] = NOT_GIVEN,
+        http_session: aiohttp.ClientSession | None = None,
+        extra_kwargs: NotGivenOr[dict[str, Any]] = NOT_GIVEN,
```

Comment:
I think depending on the model "Literal", we could make extra_kwargs a TypedDict with all the available fields.